src/CodeMirror.js

```diff
@@ -0,0 +1,218 @@
+// CodeMirror, copyright (c) by Marijn Haverbeke and others
+// Distributed under an MIT license: http://codemirror.net/LICENSE
+
+import { initScrollbars, setGuttersForLineNumbers, themeChanged, updateGutters } from "./cm_utils";
+import { Init } from "./default_options";
+import { attachDoc } from "./document_data";
+import { clearDragCursor, onDragOver, onDragStart, onDrop } from "./drop_events";
+import { on, signal, signalDOMEvent } from "./util/event";
+import { onFocus, onBlur } from "./display/focus";
+import { ensureGlobalHandlers } from "./global_events";
+import { onKeyDown, onKeyPress, onKeyUp } from "./key_events";
+import { maybeUpdateLineNumberWidth } from "./display/line_numbers";
+import { clickInGutter, onContextMenu, onMouseDown } from "./mouse_events";
+import { endOperation, operation, startOperation } from "./operations";
+import Pos from "./Pos";
+import { posFromMouse } from "./measurement/position_measurement";
+import { onScrollWheel, setScrollLeft, setScrollTop } from "./scroll_events";
+import { Range } from "./selection";
+import { extendSelection } from "./selection_updates";
+import { captureRightClick, ie, ie_version, mobile, webkit } from "./util/browser";
+import { bind, copyObj, Delayed } from "./util/misc";
+import { eventInWidget } from "./utils_widgets";
+import { e_preventDefault, e_stop } from "./util/event";
+import { clipPos } from "./utils_pos";
+import Display from "./Display";
+import Doc from "./Doc";
+
+var defaults;
+var optionHandlers;
```

Comment:
These seem like they should be imported from an options (I'd rename default_options to options, since it also holds the definition of `defineOption`) module, and only attached to `CodeMirror` in this file (along with `defineOption`)